packages/esm-patient-chart-app/src/ui-components/patient-chart.component.tsx

```diff
@@ -24,8 +25,16 @@ const PatientChart: React.FC<RouteComponentProps<PatientChartParams>> = ({ match
   const { windowSize, openWindows } = useContextWorkspace();
 
   useEffect(() => {
-    !loading && detachAll('patient-chart-workspace-slot');
-  }, [loading]);
+    const handler = () => {
```

Comment:
This is not really what we want, right?

I mean why / how should the `patientUuid` be different than the `currentPatientUuid`? If this is the case, then the original `patientUuid` is *wrong* - so this is the thing that should be fixed.

Reply:
Florian's comment makes sense to me, @donaldkibet . Can you investigate why that's happening?

Reply:
I have investigated the issue, and it seems that once you change the patient, the event listener fires first before the useEffect hook, this results in having the `patientUuid` being the previous one and the `currentPatientUuid` being the current once that has changed. subsequently the useEffect runs again since patientUuid has changed and this results in having both `patientUuid` and `currentPatientUuid` being the same.

Since this isn't the ideal situation, i have used `window.document.referrer` to get the previousPatientUuid and compare that to the `currentPatientUuid` retrieved from `location.pathname` if this two don't match that indicates that the patient details have changed and we close the opened workspace. 


```
useEffect(() => {
    const handler = () => {
      const previousPatientUuid = getPatientUuidFromUrl(window.document.referrer);
      const currentPatientUuid = getPatientUuidFromUrl(location.pathname);
      if (previousPatientUuid !== currentPatientUuid) {
        detachAll('patient-chart-workspace-slot');
      }
    };
    window.addEventListener('popstate', handler);
    return () => window.removeEventListener('popstate', handler);
  }, [patientUuid]);
```

Reply:
Okay, that makes sense. Yeah I understood your fix but it's good to have a clearer understanding of the problem. Could you link to the event listener and useEffect hook that you're talking about? Do you think there's a way we can get them to fire in the correct order?